### ISSUE TITLE
Network spam prevention

### ISSUE THREAD

**Issue by dafky2000** [tweetchain/tweetchain#9]
Need some sort of spam detection. Maybe only so many mined blocks per user, per number of confirmation blocks?

Would love ideas.

**Comment by dafky2000** [comment]
This is / could be one and the same or influenced by block content quality.

**Comment by dafky2000** [comment]
Thinking out loud: Considered spam if mined more than:

- 2 blocks out of 3
- 4 out of 10
- 6 out of 20..?

**Comment by TheGodson14** [comment]
This would be difficult, because a spammer could open multiple accounts and mine the blocks with many separate accounts. Fortunately at the time being, I don't see this as problem in the near future.

In the far future we will have to worry about it though. It would be interesting if there was a wait time restriction on the chain as well as what you mentioned. That way, when TweetCoin becomes popular, there won't be too many posts flooding Twitter.

**Comment by dafky2000** [comment]
I disagree. We need spam protection now. We already have people spamming the chain with donation addresses. This would be part of content curation and promoting content quality instead of what would be considered spam. By measuring the authenticity of an account and their tweets we could weed out spam accounts. There would be a few factors involved here.

**Comment by dafky2000** [comment]
Tweetchain.info could also play the role of only showing the last unique users block
